platform/openide.util.ui/src/org/openide/util/ImageUtilities.java

```diff
@@ -1086,29 +1049,7 @@ private static ColorModel colorModel(int transparency) {
     /**
      * Key used for composite images -- it holds image identities
      */
-    private static class CompositeImageKey {
-        Image baseImage;
-        Image overlayImage;
-        int x;
-        int y;
-
-        CompositeImageKey(Image base, Image overlay, int x, int y) {
-            this.x = x;
-            this.y = y;
-            this.baseImage = base;
-            this.overlayImage = overlay;
-        }
-
-        @Override
-        public boolean equals(Object other) {
-            if (!(other instanceof CompositeImageKey)) {
-                return false;
-            }
-
-            CompositeImageKey k = (CompositeImageKey) other;
-
-            return (x == k.x) && (y == k.y) && (baseImage == k.baseImage) && (overlayImage == k.overlayImage);
-        }
+    private record CompositeImageKey(Image baseImage, Image overlayImage, int x, int y) {
```

Comment:
Switching this to record will change the comparison of Image from identity comparison to comparison using equals(). That may be fine since BufferedImage does not actually define equals or hashCode. But then we should remove the custom hashCode() implementation as well for consistency.

Reply:
good catch - will add the custom equals again. I kept it originally but when I looked through it today I must have overlooked that the images were compared with `==`.